src/auth/models/strategy.class.ts

```diff
@@ -256,14 +258,41 @@ export abstract class Strategy extends events.EventEmitter {
 
     public makeAuthorization = (passport: any) => `Bearer ${passport.user.tokenset.accessToken}`
 
-    public setHeaders = (req: Request, res: Response, next: NextFunction): void => {
+    public setHeaders = async (req: Request, res: Response, next: NextFunction): Promise<void> => {
         if (req.session?.passport?.user) {
-            req.headers['user-roles'] = req.session.passport.user.userinfo.roles.join()
-            req.headers.Authorization = this.makeAuthorization(req.session.passport)
+            if (
+                this.options.routeCredential &&
+                this.options.routeCredential.routes &&
+                this.options.routeCredential.routes.includes(req.url)
+            ) {
+                let routeCredentialToken
+                if (req.session.routeCredentialToken) {
+                    routeCredentialToken = req.session.routeCredentialToken
```

Comment:
What if the token has expired? 

Also it is being stored in the user's session, so each user has to login individually as the system user. Should be stored globally, not at the user session level so that there is only one login.

Reply:
Agree with you. I will move it to applicationcache. Thanks